### ISSUE TITLE
Psalm is emitting PossiblyUndefinedVariable for array access of loop value variable

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#374]
Expected: No violation is emitted
Observed: PossiblyUndefinedVariable on $value, which is guaranteed to be defined.

```php
<?php

/**
 * @return array
 */
function get_config() : array {
    return [['enabled' => true]];
}

class tag_util_nav {
    public static function get_settings() : array {
        $result = array();
        foreach (get_config() as $key => $value) {
            if ($value['enabled']) {  // unexpected PossiblyUndefinedVariable - Found a possibly undefined variable when evaluating $value['enabled']

                $result[] = $key;
            }   
        }   

        return $result;
    }   
}
```

**Comment by weirdan** [comment]
Wouldn't that require proving that all of the entries returned from get_config() have 'enabled' key? Sounds like expecting too much from the type inference.

Adding `@psalm-return array<int,array{enabled:bool}>` clears the error.

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/d10898b853b6e3f264c9477cab07a9a13de01f58

Sorry
